Source/Dafny/Compiler-Csharp.cs

```diff
@@ -347,7 +347,8 @@ public CsharpCompiler(ErrorReporter reporter)
       //    }}
       //   ...
       // }
-      var DtT_TypeArgs = TypeParameters(dt.TypeArgs);
+      var nonGhostTypeArgs = SelectNonGhost(dt, dt.TypeArgs);
```

Comment:
I feel we need a compilation step that removes anything ghost, to prevent having to filter out ghostness in each of the compilers, but I guess that's out of scope for this PR.

Reply:
I agree, on both accounts.